stackit/internal/services/postgresql/instance/resource.go

```diff
@@ -122,7 +122,16 @@ func (r *instanceResource) Configure(ctx context.Context, req resource.Configure
 // Schema defines the schema for the resource.
 func (r *instanceResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
 	descriptions := map[string]string{
-		"main":        "PostgreSQL instance resource schema. Must have a `region` specified in the provider configuration.",
+		"main": "PostgreSQL instance resource schema. Must have a `region` specified in the provider configuration.",
+		"deprecation_message": strings.Join(
+			[]string{
+				"The STACKIT PostgreSQL service will reach its end of support on June 30th 2024.",
+				"Resources of this type will stop working after that.",
+				"Use stackit_postgresqlflex_instance instead.",
```

Comment:
Just noticed, the name of the resource is `stackit_postgresflex_...` and not `stackit_postgresqlflex_...`